onnxruntime/core/providers/dml/OperatorAuthorHelper/OperatorHelper.cpp

```diff
@@ -614,11 +615,18 @@ namespace OperatorHelper
             ML_CHECK_VALID_ARGUMENT(dimensionCount > 2,
                 "FusedMatMul operator: Tensor size should be more than 2, if attribute transBatch is true");
 
-            std::rotate(newSizes.begin(), newSizes.end() - 2, newSizes.end() - 1);
-            std::rotate(newStrides.begin(), newStrides.end() - 2, newStrides.end() - 1);
+            uint32_t leadingDim = newSizes[0];
+            uint32_t leadingStride = newStrides[0];
+            for (size_t i = 0; i < newSizes.size() - 2; ++i)
+            {
+                newSizes[i] = newSizes[i + 1];
+                newStrides[i] = newStrides[i + 1];
+            }
+            newSizes[newSizes.size() - 2] = leadingDim;
+            newStrides[newStrides.size() - 2] = leadingStride;
```

Comment:
I realize the beginning parameter of the new vector was in-correct. Can we keep using `std::rotate` with updated parameter, if it makes sense? 
`std::rotate(newSizes.begin(), newSizes.begin() + 1, newSizes.end() - 1);`
[I have verified with a small code.]

![image](https://user-images.githubusercontent.com/11188170/211630457-f0e697ab-241d-4da8-9ada-5eff251ec750.png)